### DISCUSSION TITLE
Any tutorial or detailed docs available?

### DISCUSSION THREAD

**Discussion by Mohsentrade741** [ulixee/unblocked#85] [Q&A]
Hello
I just found this project which looks amazing, however as a beginner I can not figure out how to use it. And the doc is so limited.
Is there any video tutorial or doc to start with?
Can I use this as a multi account solution with different profiles? And do I need to use it with the Ulixee main platform or can it be used with Puppeteer or alone?
Thanks in advance for any help!

**Comment by blakebyrnes** [comment]
Thanks! The simplest way is to use Hero. This project is sort of the internal flow for building hero, but it got so big that it needed its own project. There's more docs at ulixee.org. Let me know if that's helpful

**Reply by Mohsentrade741** [reply]
Hi
Thanks for the reply.
In fact I saw the hero docs, however it seems the hero is headless browser while I need to use this as normal browser.
One amazing part I see is the great set of profiles in profile-data folder. The question is that if I can use those profiles or some with normal Chrome/Chromium or other browsers? I searched a lot but couldn't find a way to use those profiles.
Apart from the Browserstack profiles which seems that can be used with cloud service of Browserstack, do you think the rest can be used with local instance of browsers, if so could you please give me some hints how they can be imported/used with normal browsers.
Thanks again!

**Reply by blakebyrnes** [reply]
Ah, I see. Those are kind of the details behind the scenes of how we're generating profiles to be used with Chrome inside of Hero. We have a number of people launching Hero headed (it's a launch config) and navigating around with Hero once they do so (it was an unexpected use case). The profiles and code is all there, but isn't really setup to be used to drive Chrome without the underlying engine (Hero/unblocked). Chrome is the only browser that the profiles can drive at the moment. We had worked on Safari for a while, but it it turns out the automated version is very different from the actual safari browser, and the volume is too low to justify the work.

**Reply by Mohsentrade741** [reply]
It's a shame that such good work hasn't got a youtube channel with some examples. Surprisingly I couldn't find much details apart from your own website and Github repository.
I will try to see if I can run the Hero. If so, using it with the profiles and also a proxy perhaps would be a great multilogin solution.

**Reply by blakebyrnes** [reply]
Thanks for the praise :) We'll get to better documentation once we're further on the Ulixee platform, which is the full vision (and eating up so much of our time). If you want to share your findings on youtube ... 🥇

**Reply by Mohsentrade741** [reply]
Hello again!
I have been trying to run the Hero. First I had some errors (deprecation warning) on a custom version of windows 10 (ReviOS) but finally it worked on another windows 10. Here are some questions please:
1: In the Hero doc says some fingerprints like audio and webgl are in "to do" list as I see webgl and audio fingerprints in json files in Browser-profile-data folder just wanted to make sure that the doc is updated and correct or not.
2: Apart from the default-emulator is there any other available or can we build custom one using the "emulator builder" in the Unblocked repo? Is there any doc available for the builder? (it doesn't have a readme too) Do you think using a custom emulator for one of the profiles in the Browser-Profiles-data directory would include the audio and webgl too? Or could those profiles be used with Hero directly? as I'm not sure if the Hero and Unblocked are the same in term of fingerprints.

Thanks for your time!

**Reply by blakebyrnes** [reply]
1. The audio and webgl still doesn't do anything to modify fingerprints, but this is only a problem if you use the same ip at scale.
2. The builder is really used only to generate the default-emulator. It's not built with a general purpose user in mind at all. I think you're thinking the profile data is different than what it is. It's currently just a measurement of how browsers react to various scenarios defined in double agent. It doesn't come with anything about what to do with that data. We're using the chrome profiles to then test creating an automated browser that tries to make itself look like those profiles. Hero uses unblocked's default browser emulator underneath. Unblocked is just the "internal" workflow of creating the default browser emulator and automated agent. Hero is the driver with a nicer api to use as a client consumer of it. 

All that said, you're obviously welcome to build whatever you want out of the other profiles! That's just what's here right now.
